### PR TITLE
feat: add the lists component

### PR DESCRIPTION
**Pull request by devodii**
Hey, I added the List components. (I would rename to Lists in the next commit)
What props does it receive? 🤔 
- An array of contents that are grouped as `list`

Next, I'd accept props to show the list-style-type
What other props would we need for this?

### CODE REVIEW

File: components/List.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+interface Props {
+  lists: { content: string }[];
+}
```
Comment (BleedingDev) [inline components/List.tsx:3]: **praise:** We should have Prettier formatting turned on automatically. Later I will add it as a pre-commit hook.

File: components/List.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+export function List({ lists }: Props) {
+  const className = "grid grid-cols-1 gap-1 list-decimal";
```
Comment (BleedingDev) [inline components/List.tsx:5]: **praise:** We should add classnames directly to the element, not separate it from component.

File: components/List.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  return (
+    <ul className={className}>
```
Comment (BleedingDev) [inline components/List.tsx:7]: **praise:** We should switch between <ul> and <ol> based on the prop `ordered` which is `boolean`.

File: components/List.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+      {lists.map((list, idx) => {
```
Comment (BleedingDev) [inline components/List.tsx:8]: **praise:** We should not use order in the list as a key for the component. It is not recommended. Probably the best way (here) is to hash or encode each item unless we generate id for each item.

File: components/List.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+        const { content } = list;
```
Comment (BleedingDev) [inline components/List.tsx:9]: **praise:** We should use destructuring directly in the map and return pure component (e.g. no `{ return ... }`). Example:

`lists.map(({content}) => <li>{content}</li>)`

File: puck.config.tsx [hunk 6]
```diff
@@ -10,7 +10,7 @@ import { Video } from "components/Video";
 import { Root } from "./components/Root";
 import { Typography } from "./components/Typography";
 import { Divider } from "components/Divider";
-
+import { List } from "components/List";
```
Comment (BleedingDev) [inline puck.config.tsx:13]: **praise:** We should have empty line after imports (but this handles Prettier, do you have it working?).

**Review by BleedingDev — CHANGES_REQUESTED** [review]

File: components/List.tsx (hunk 1 above)
Comment (devodii) [reply components/List.tsx:3]: What line isn't correctly formatted?

File: components/List.tsx (hunk 4 above)
Comment (devodii) [reply components/List.tsx:8]: Now, we're receiving the data without an id. do i use the hashed content?

File: components/List.tsx (hunk 3 above)
Comment (devodii) [reply components/List.tsx:7]: We should switch between what? I didnt see anything there.

File: puck.config.tsx (hunk 6 above)
Comment (devodii) [reply puck.config.tsx:13]: It been working. I just didn't save the file before commiting.

File: components/List.tsx (hunk 1 above)
Comment (BleedingDev) [reply components/List.tsx:3]: I think it is fine on the second look. Gonna add Prettier today as a hook.

File: components/List.tsx (hunk 4 above)
Comment (devodii) [reply components/List.tsx:8]: Okay

File: components/List.tsx (hunk 1 above)
Comment (devodii) [reply components/List.tsx:3]: Okay

Comment (BleedingDev) [reply components/List.tsx:6]: This solution is way too complex. There is no need to create separate component. I will open PR to your branch. :)

Comment (devodii) [reply components/List.tsx:6]: What do i remove?

**Review by BleedingDev — APPROVED** [review]

**Review by BleedingDev — APPROVED** [review]

**Review by BleedingDev — APPROVED** [review]

**Pull request merged**